web3-react/src/example/Example.tsx

```diff
@@ -0,0 +1,58 @@
+import './Example.css'
+
+import { useWeb3React } from '@web3-react/core'
+import React, { useEffect, useState } from 'react'
+
+import { CurrentConfig } from '../config'
+import { ConnectionOptions } from '../libs/components/ConnectionOptions'
+import { ConnectionType, switchNetwork } from '../libs/connections'
+import { CHAIN_INFO } from '../libs/constants'
+
+// Listen for new blocks and update the wallet
+const useOnBlockUpdated = (callback: (blockNumber: number) => void) => {
+  const { provider } = useWeb3React()
+  useEffect(() => {
+    if (!provider) {
+      return
+    }
+    const subscription = provider.on('block', callback)
+    return () => {
+      subscription.removeAllListeners()
+    }
+  })
+}
+
+const Example = () => {
+  const { chainId, account, isActive } = useWeb3React()
+  const [blockNumber, setBlockNumber] = useState<number>(0)
+  const [connectionType, setConnectionType] = useState<ConnectionType | null>(null)
+
+  // Listen for new blocks and update the wallet
+  useOnBlockUpdated((blockNumber: number) => {
+    setBlockNumber(blockNumber)
+  })
+
+  return (
+    <div className="App">
+      {CurrentConfig.rpc.mainnet === '' && <h2 className="error">Please set your mainnet RPC URL in config.ts</h2>}
```

Comment:
Will this work when I switch network to Polygon (or something different than a `mainnet`)? I think this check shouldn't be hardcoded to `mainnet` but a currently selected network instead.

Reply:
Good call!